### PR TITLE
[WIP][DNM] Nomifactory - The Industrial Era

### PR DESCRIPTION
**Pull request by ZephyrTFA**
## About The Pull Request

Converts stock part creation into a non-linear and involved system in order to: slow it down, make it more rewarding to optimize, slow it the fuck down.
This system also supports more recipes than just stock parts, so consider making your brand new items created in a nomifactory recipe, I tried to make it as easy as possible!

## Todo
 - [x] - Base logic
 - [x] - Multi-tile machines/logic
 - [ ] - Sprites
 - [x] - Base Recipes
 - [ ] - Additional Recipes
 - [ ] - Remove old recipes from auto/proto lathes
 - [x] - Upgrading Machines

## Why It's Good For The Game

Stock parts, and most things made from the lathe, are both boring and non-engaging. Its quite literally place a lathe and it just works.

### CODE REVIEW

**Comment by Stalkeros2** [comment]
OH MY GOD INDUSTRIAL REVOLUTION WOOOOOOO
Can't wait to cannibalise this code to make custom item factories.

File: code/modules/nomifactory/deep_miner.dm [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		construction_stage--
+		return
+
+	if(progress++ > payout_progress)
+		progress = 0
+		payout()
+
+/obj/structure/nomifactory/deep_miner/proc/create_output_map()
+	return list(
+		/obj/item/stack/ore/glass = 50
```
Comment (ZephyrTFA) [inline code/modules/nomifactory/deep_miner.dm]:
```suggestion
		/obj/item/stack/ore/glass = 50,
```
hsss

**Comment by ZephyrTFA** [comment]
Multi tile machinery should now work, I'm unsure if I want all the parts to be able to be used to set the recipe individually and/or accept input/output. TBD at a later date

**Comment by ZephyrTFA** [comment]
Due to certain real life issues I lack the mental capacity to even attempt to finish this. However, for the most part it is completely done. The only things currently needed for this is spritework, recipe creation, and removal of the old recipes for stock parts.
If someone could pick this up for me that would be grand; if not it's just a rather big waste of effort.

**Comment by ZephyrTFA** [comment]
> @ZephyrTFA Once I come back with commits should I PR to your branch or create a new PR to shiptest?

My branch on my repo

**Comment by ZephyrTFA** [comment]
cope + ratio

**Comment by ZephyrTFA** [comment]
I'm lazy and have too many project + this can wait until after research rework + cope + ratio

**Pull request closed without merging**